api/src/test/java/org/pmiops/workbench/db/dao/InstitutionDaoTest.java

```diff
@@ -0,0 +1,69 @@
+package org.pmiops.workbench.db.dao;
+
+import static com.google.common.truth.Truth.assertThat;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.pmiops.workbench.db.model.DbInstitution;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;
+import org.springframework.dao.DataIntegrityViolationException;
+import org.springframework.test.context.junit4.SpringRunner;
+
+@RunWith(SpringRunner.class)
+@DataJpaTest
+public class InstitutionDaoTest {
+  @Autowired InstitutionDao institutionDao;
+
+  @Test
+  public void testDao() {
+    final DbInstitution testInst = new DbInstitution("Broad", "The Broad Institute");
+    institutionDao.save(testInst);
+    assertThat(institutionDao.findOneByApiId("Broad")).isEqualTo(testInst);
+    assertThat(institutionDao.findAll()).hasSize(1);
+
+    // update existing entity, don't change size
+
+    testInst.setApiId("Verily");
+    institutionDao.save(testInst);
+    assertThat(institutionDao.findAll()).hasSize(1);
+    assertThat(institutionDao.findOneByApiId("Verily")).isEqualTo(testInst);
+
+    testInst.setLongName("Yea, Verily");
```

Comment:
Cool, but might be a more robust test if the short name isn't a substring of the long name.